### PR TITLE
Add V3 Support

### PR DESCRIPTION
**Pull request by Oscaarjs**
This is a continuation of  #548 so credit to its contributors and author [@stillmatic](https://github.com/stillmatic).

I've tried to adress some of the comments recieved on that PR.

Potential TODOs:

- Dynamic _LANGUAGE_CODES in tokenizer.py depending on whether V3 (which supports "yue") is loaded.

### CODE REVIEW

**Comment by Oscaarjs** [comment]
@nguyendc-systran , would be great if you could have a look

File: faster_whisper/utils.py [hunk 1]
```diff
@@ -76,6 +77,7 @@ def download_model(
 
     allow_patterns = [
         "config.json",
+        "preprocessor_config.json",
```
Comment (trungkienbkhn) [inline faster_whisper/utils.py:80]: @Oscaarjs , hello.  We have generated the Systran Whisper large-v3 conversion model with new file preprocessor_config.json in the HF-to-ct2 script, could you please update this info also in README.md file of your PR.
Example: ct2-transformers-converter --model openai/whisper-large-v3 --output_dir whisper-large-v2-ct2 \
    --copy_files tokenizer.json **preprocessor_config.json** --quantization float16

Comment (Oscaarjs) [reply faster_whisper/utils.py:80]: @trungkienbkhn Ah yes! Did the changes, please check if I made them correctly

**Comment by salahzoubi** [comment]
@Oscaarjs im guessing this still doesn't allow for batch transcribe (which is built-in large-v3)?

**Comment by Oscaarjs** [comment]
> @Oscaarjs im guessing this still doesn't allow for batch transcribe (which is built-in large-v3)?

What do you mean by built-in? Afaik there's nothing changed with v3 and v2 that changes that part of the model. Or are you referring to HFs pipeline implementation of it? (which do support batching)

**Comment by funboarder13920** [comment]
> @Oscaarjs im guessing this still doesn't allow for batch transcribe (which is built-in large-v3)?

Your question was off topic in the other PR, it is still off topic in this PR.
Batch inference has nothing to do with large-v3

**Review by trungkienbkhn — APPROVED** [review]

File: faster_whisper/utils.py (hunk 1 above)
Comment (trungkienbkhn) [reply faster_whisper/utils.py:80]: Sounds good to me. Hopefully there will be some benchmark tests for the faster whisper large-v3 soon.

**Review by nguyendc-systran — APPROVED** [review]

**Pull request merged**